queries.sql

```diff
@@ -29,3 +29,86 @@ WHERE name <> 'Gabumon';
 SELECT *
 FROM animals
 WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+------------------------------------ROLLBACK the changes----------------------------------------------
+
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+-----------------------------------Update the species and COMMIT--------------------------------------
+
+BEGIN; 
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+SELECT * FROM animals WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+SELECT * FROM animals WHERE species = 'pokemon';
+
+COMMIT;
+
+SELECT * FROM animals;
+
+
+
+-------------------------------------DELETE and ROLLBACK-------------------------------------------
+
+BEGIN; 
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals
```

Comment:
Hello,

Great job on your SQL submission! 🌟 It's evident that you've understood the core concept of transactions and how to roll them back, which is crucial in maintaining database integrity. Your code is clear, and the logic is sound. Well done!

However, there are some improvements needed to fully align with the project requirements.

### Suggestions 🧐

1. **Verification of Deletion:**
   - After executing the `DELETE` command, insert a `SELECT` statement to validate whether all records from the `animals` table have indeed been deleted.
     ```sql
     SELECT * FROM animals; -- This should return no rows, verifying the deletion.
     ```

### Corrected Sequence 🛠️

Here’s how the enhanced sequence should look like:

```sql
BEGIN; 

DELETE FROM animals;

-- Verification of Deletion
SELECT * FROM animals; -- Should return no rows, signifying successful deletion.

ROLLBACK;

-- Verification of Rollback
SELECT * FROM animals; -- Should return all initial rows, proving successful rollback.
```

### Conclusion 🌟

You've done exceptionally well, and with a few tweaks, your submission will be spot on! Implementing these suggestions ensures not only adherence to project requirements but also solidifies your understanding of SQL transactions and their implications. Keep up the excellent work, and remember, the journey of coding is filled with continuous learning and improvement! 👏

Reply:
![Uploading delete22.png…]()
 done.

Reply:
![delete22](https://github.com/coker2019/vet-clinic/assets/47523191/b7eadc22-d1ce-418c-a9dd-da866db3c793)

Reply:
@coker2019, Thank you for implementing the suggested changes swiftly! 🌟 I’ve reviewed your corrected submission, and it now aligns perfectly with the project requirements. Well done! 👏